### PR TITLE
adding 9999 ebuilds for kicad

### PR DESCRIPTION
**Pull request by eeyrjmr**
v5.99 has been in development for a couple of years and rc1 is due end of november and v6.0.0 around xmas 2021. 
These ebuilds include all the tweaks to the build as the development progresses either from gentoo (thread removed from boost) or kicad (github remove, oce depreciated, plugin manager added).

### CODE REVIEW

**Review by juippis — COMMENTED** [review]
@zpuskas any words? These'd be maintained by you.

I'm personally against adding these ebuilds, and would rather package the _rc etc versions unkeyworded for example. We can never guarantee a -9999 builds at any given time. And it's up to the maintainer to keep versioned and -9999 ebuilds in sync.

**Comment by eeyrjmr** [comment]
@juippis While I understand the sentiment, the 9999 ebuilds are useful for those of us contributing to Kicad (or testing).
However, the reason pushing these 9999 here was because I have been keeping them up to date with regards to changes (dropping github, dropp oce, add plugin manager etc...) and rc1 was due to be released

RC1 was released a few days ago so these can be copied to 6.0.0_rc1 and be viable for in-tree

**Comment by thesamesam** [comment]
Would you mind fixing the CI errors (let me know if you need help) and chucking in the RC1 too, then I think we can do it?

I'm always quite surprised but Blender, kicad, stuff like that usually has a lot of fans of live rebuilds / RCs etc, I guess because of an unusual development pace or something.

**Comment by eeyrjmr** [comment]
NOTE: the QA warning **USE flag 'oce' missing from packages:** is no longer valid. KiCad drop OCE support in October 2021 ( https://www.mail-archive.com/kicad-developers@lists.launchpad.net/msg39760.html) hence the removal of the USE flag

**Comment by thesamesam** [comment]
> NOTE: the QA warning **USE flag 'oce' missing from packages:** is no longer valid. KiCad drop OCE support in October 2021 ( https://www.mail-archive.com/kicad-developers@lists.launchpad.net/msg39760.html) hence the removal of the USE flag

Ah, we need to update the meta and the other package to have oce(-) then (“if the flag isn’t there, assume it’s off”). Will add a link when not on mobile but it’s “use defaults” in the dev manual.

File: sci-electronics/kicad-symbols/kicad-symbols-9999.ebuild [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+fi
+
+LICENSE="CC-BY-SA-4.0"
+SLOT="0"
+IUSE=""
+
+DEPEND=""
+RDEPEND=">=sci-electronics/kicad-5.1.0"
+
+S="${WORKDIR}/${PN}-${MY_PV}"
```
Comment (juippis) [inline sci-electronics/kicad-symbols/kicad-symbols-9999.ebuild:30]: Wow, a lot has happened here!

I haven't written live ebuilds and don't know the interactions, but I believe this `S` needs to be defined in the `if-else` block above.
Basically any -9999 ebuilds fail to:
```
>>> Source unpacked in /var/tmp/portage/sci-electronics/kicad-footprints-9999/work
 * ERROR: sci-electronics/kicad-footprints-9999::gentoo failed (prepare phase):
 *   The source directory '/var/tmp/portage/sci-electronics/kicad-footprints-9999/work/kicad-footprints-' doesn't exist
 * 
 * Call stack:
 *            ebuild.sh, line  809:  Called __ebuild_main 'prepare'
 *   phase-functions.sh, line 1059:  Called __dyn_prepare
 *   phase-functions.sh, line  385:  Called die
 * The specific snippet of code:
 *   		die "The source directory '${S}' doesn't exist"
```

```
/var/tmp/portage/vbslogs/kicad-meta-9999-0.json:        "exit_code": 1,
/var/tmp/portage/vbslogs/kicad-meta-9999-0.json:        "exit_code": 1,
/var/tmp/portage/vbslogs/kicad-meta-9999-0.json:        "exit_code": 1,
/var/tmp/portage/vbslogs/kicad-meta-9999-0.json:        "exit_code": 1,
/var/tmp/portage/vbslogs/kicad-symbols-9999-3.json:        "exit_code": 1,
/var/tmp/portage/vbslogs/kicad-symbols-9999-3.json:        "exit_code": 1,
/var/tmp/portage/vbslogs/kicad-templates-9999-2.json:        "exit_code": 1,
/var/tmp/portage/vbslogs/kicad-templates-9999-2.json:        "exit_code": 1,
/var/tmp/portage/vbslogs/kicad-9999-1.json:        "exit_code": 1,
/var/tmp/portage/vbslogs/kicad-9999-1.json:        "exit_code": 1,
/var/tmp/portage/vbslogs/kicad-9999-1.json:        "exit_code": 1,
/var/tmp/portage/vbslogs/kicad-9999-1.json:        "exit_code": 1,
/var/tmp/portage/vbslogs/kicad-9999-1.json:        "exit_code": 1,
/var/tmp/portage/vbslogs/kicad-9999-1.json:        "exit_code": 1,
/var/tmp/portage/vbslogs/kicad-9999-1.json:        "exit_code": 1,
/var/tmp/portage/vbslogs/kicad-footprints-9999-6.json:        "exit_code": 1,
/var/tmp/portage/vbslogs/kicad-footprints-9999-6.json:        "exit_code": 1,
Grepping through the logs for any errors or QA notices...
/var/tmp/portage/vbslogs/build/sci-electronics/kicad-9999:20211122-132136.log: * ERROR: sci-electronics/kicad-9999::gentoo failed (prepare phase):
/var/tmp/portage/vbslogs/build/sci-electronics/kicad-9999:20211122-132204.log: * ERROR: sci-electronics/kicad-9999::gentoo failed (prepare phase):
/var/tmp/portage/vbslogs/build/sci-electronics/kicad-9999:20211122-132152.log: * ERROR: sci-electronics/kicad-9999::gentoo failed (prepare phase):
/var/tmp/portage/vbslogs/build/sci-electronics/kicad-9999:20211122-132143.log: * ERROR: sci-electronics/kicad-9999::gentoo failed (prepare phase):
/var/tmp/portage/vbslogs/build/sci-electronics/kicad-9999:20211122-132128.log: * ERROR: sci-electronics/kicad-9999::gentoo failed (prepare phase):
/var/tmp/portage/vbslogs/build/sci-electronics/kicad-templates-9999:20211122-133340.log: * ERROR: sci-electronics/kicad-templates-9999::gentoo failed (prepare phase):
/var/tmp/portage/vbslogs/build/sci-electronics/kicad-9999:20211122-132158.log: * ERROR: sci-electronics/kicad-9999::gentoo failed (prepare phase):
/var/tmp/portage/vbslogs/build/sci-electronics/kicad-9999:20211122-132210.log: * ERROR: sci-electronics/kicad-9999::gentoo failed (prepare phase):
/var/tmp/portage/vbslogs/build/sci-electronics/kicad-footprints-9999:20211122-135501.log: * ERROR: sci-electronics/kicad-footprints-9999::gentoo failed (prepare phase):
/var/tmp/portage/vbslogs/build/sci-electronics/kicad-symbols-9999:20211122-133350.log: * ERROR: sci-electronics/kicad-symbols-9999::gentoo failed (prepare phase):
/var/tmp/portage/vbslogs/build/sci-electronics/kicad-symbols-9999:20211122-133408.log: * ERROR: sci-electronics/kicad-symbols-9999::gentoo failed (prepare phase):
/var/tmp/portage/vbslogs/build/sci-electronics/kicad-templates-9999:20211122-133319.log: * ERROR: sci-electronics/kicad-templates-9999::gentoo failed (prepare phase):
/var/tmp/portage/vbslogs/build/sci-electronics/kicad-footprints-9999:20211122-135511.log: * ERROR: sci-electronics/kicad-footprints-9999::gentoo failed (prepare phase):
```

On a positive note, the _rc ones seem to work!

Comment (eeyrjmr) [reply sci-electronics/kicad-symbols/kicad-symbols-9999.ebuild:30]: Damn... I had been working with the 9999 for months now and then refactored to get rc1

I'll do the of/else for S as this isn't needed for 9999 as you said

**Comment by ezzieyguywuf** [comment]
@thesamesam  @juippis  I took a stab at cleaning up the commit history (this is imperfect but should be a good starting point). Hopefully this helps reduce your workload on this PR a bit - I could not find a clean way to submit a PR to @eeyrjmr 's repo since I ended up rewriting history to clean things up.

[here](https://github.com/ezzieyguywuf/gentoo/tree/update_kicad) is the branch on my fork that contains the cleaned up commits.

**Review by juippis — COMMENTED** [review]
Also I'd appreciate if you bumped EAPI to 8, but it's not super important right now.

I still didn't test each package yet (and didn't get to the CI error yet) but I feel like it's faster if I / sam fix the remaining errors, but by all means if you want to work on this let us know and we'll continue! Testing these is getting a bit painful due to the non-matching git history :P 

@ezzieyguywuf it is great help, but needs to be kept in-sync with this PR to be useful. Also you can use git commit --author even in your own branch (that's not a problem now, can easily change it while merging).

**Comment by thesamesam** [comment]
Just for anyone following: I ended up merging this a bit early accidentally (while I was doing some checking & fixups) but this is now merged! I've made some minor changes since to fix tests and stuff.

Thanks all!

**Pull request closed without merging**